### PR TITLE
Migrate to Rails 5 on Heroku

### PR DESCRIPTION
**Pull request by mattharris5**

### CODE REVIEW

**Comment by mattharris5** [comment]
Lots of work to do before this can be merged...

File: app/controllers/session_controller.rb [hunk 1]
```diff
@@ -5,9 +5,9 @@ class SessionController < ApplicationController
   before_filter :apply_customer_styles
 
   def new
-    redirect_to locator_url(subdomain: false) if Customer.current_customer.nil? || Customer.current_customer.new_record?
-    redirect_to "/auth/#{Customer.allowable_login_methods_list.first}" if Customer.allowable_login_methods_list.size == 1 && session[:external_login_context] != :rsvp
-    render layout: "basic"
+    redirect_to locator_url(subdomain: false) and return if Customer.current_customer.nil? || Customer.current_customer.new_record?
```
Comment (mattharris5) [inline app/controllers/session_controller.rb]: I don't think we need this... Was the login page not working without this? Also be sure to keep things like this separated into a different commit since I don't think this has anything to do with searchkick.

File: config/initializers/api_keys.rb [hunk 2]
```diff
@@ -1,4 +1,2 @@
-s3 = Aws::S3::Client.new
-resp = s3.get_object( bucket: ENV['S3_BUCKET_NAME'], key: ENV['API_KEYS_PATH'] )
-API_KEYS = YAML::load(resp.body.read)
-Rails.logger.info { "Successfully loaded keys from S3" }
+config_file_path = File.join(ENV['SHARED_CONFIG_ROOT'] || "#{Rails.root}/config", "api-keys.yml")
```
Comment (mattharris5) [inline config/initializers/api_keys.rb]: Please be very careful about making changes like this!! This is going to be break the app, so I'll revert it.

File: app/controllers/session_controller.rb (hunk 1 above)
Comment (aayush19saxena) [inline app/controllers/session_controller.rb]: Hey Matt, I was getting a double render error at the login page. "Render and/or redirect were called multiple times in this action. Please note that you may only call render OR redirect, and at most one time)" sorry about the commit.

Comment (mattharris5) [inline app/controllers/session_controller.rb]: Interesting. I will have to look at that and figure out why it wasn't showing that for me. If it's double rendering then that means we have some other condition that's not hitting the existing `if` statements, I think.
